nf_core/modules/modules_json.py

```diff
@@ -1185,7 +1247,7 @@ def recreate_dependencies(self, repo, org, subworkflow):
 
         sw_path = Path(self.subworkflows_dir, org, subworkflow)
         dep_mods, dep_subwfs = get_components_to_install(sw_path)
-
+        assert self.modules_json is not None  # mypy
```

Comment:
the problem is that mypy doesn't now when which function is called when and since we initialize it as none before we load the modules_json file, mypy assumes it can still be none (even though it isn't, therefore this shouldn't fail)